### PR TITLE
Drop qjsonwrapper dependency

### PR DESCRIPTION
**Pull request by nyalldawson**
Rework the oauth2 config to/from json conversion to avoid QMetaObject introspection, as this is fragile and prone to breakage without any compilation warning or errors. This allows us to remove the qjsonwrapper dependency, and drop an unwanted embedded library.

### CODE REVIEW

**Comment by nyalldawson** [comment]
FYI @rouault

File: src/auth/oauth2/core/qgsauthoauth2config.cpp [hunk 1]
```diff
@@ -319,29 +320,43 @@ bool QgsAuthOAuth2Config::loadConfigTxt(
...
-      // safety check -- qvariant2qobject asserts if an non-matching property is found in the json
-      for ( QVariantMap::const_iterator iter = variantMap.constBegin(); iter != variantMap.constEnd(); ++iter )
-      {
-        const QVariant property = this->property( iter.key().toLatin1() );
-        if ( !property.isValid() ) // e.g. not a auth config json file
-          return false;
-      }
 
-      QJsonWrapper::qvariant2qobject( variantMap, this );
+      setApiKey( variantMap.value( QStringLiteral( "apiKey" ) ).toString() );
```
Comment (rouault) [inline src/auth/oauth2/core/qgsauthoauth2config.cpp:337]: what happens if the variantMap lacks one of the item ? Wouldn't it be preferable not to call the set method then ?

File: src/auth/oauth2/core/qgsauthoauth2config.cpp [hunk 2]
```diff
@@ -319,29 +320,43 @@ bool QgsAuthOAuth2Config::loadConfigTxt(
...
+      setClientId( variantMap.value( QStringLiteral( "clientId" ) ).toString() );
+      setClientSecret( variantMap.value( QStringLiteral( "clientSecret" ) ).toString() );
+      setConfigType( static_cast<ConfigType>( variantMap.value( QStringLiteral( "configType" ) ).toInt() ) );
```
Comment (rouault) [inline src/auth/oauth2/core/qgsauthoauth2config.cpp:340]: this is undefined behaviour if the integer value is out of range of the enumeration

File: src/auth/oauth2/core/qgsauthoauth2config.cpp [hunk 3]
```diff
@@ -319,29 +320,43 @@ bool QgsAuthOAuth2Config::loadConfigTxt(
...
+      setDescription( variantMap.value( QStringLiteral( "description" ) ).toString() );
+      setGrantFlow( static_cast<GrantFlow>( variantMap.value( QStringLiteral( "grantFlow" ) ).toInt() ) );
```
Comment (rouault) [inline src/auth/oauth2/core/qgsauthoauth2config.cpp:342]: same here

File: src/auth/oauth2/core/qgsauthoauth2config.cpp [hunk 4]
```diff
@@ -319,29 +320,43 @@ bool QgsAuthOAuth2Config::loadConfigTxt(
...
+      setId( variantMap.value( QStringLiteral( "id" ) ).toString() );
+      setName( variantMap.value( QStringLiteral( "name" ) ).toString() );
+      setPassword( variantMap.value( QStringLiteral( "password" ) ).toString() );
+      setPersistToken( variantMap.value( QStringLiteral( "persistToken" ) ).toBool() );
+      setQueryPairs( variantMap.value( QStringLiteral( "queryPairs" ) ).toMap() );
+      setRedirectHost( variantMap.value( QStringLiteral( "redirectHost" ) ).toString() );
+      setRedirectPort( variantMap.value( QStringLiteral( "redirectPort" ) ).toInt() );
+      setRedirectUrl( variantMap.value( QStringLiteral( "redirectUrl" ) ).toString() );
+      setRefreshTokenUrl( variantMap.value( QStringLiteral( "refreshTokenUrl" ) ).toString() );
+      setAccessMethod( static_cast<AccessMethod>( variantMap.value( QStringLiteral( "accessMethod" ) ).toInt() ) );
```
Comment (rouault) [inline src/auth/oauth2/core/qgsauthoauth2config.cpp:352]: same here

File: src/auth/oauth2/core/qgsauthoauth2config.cpp (hunk 2 above)
Comment (nyalldawson) [reply src/auth/oauth2/core/qgsauthoauth2config.cpp]: Ok, addressed!

**Comment by nyalldawson** [comment]
@rouault all addressed!

**Review by rouault — APPROVED** [review]

**Pull request merged**